src/mixins/itext_key_behavior.mixin.js

```diff
@@ -277,39 +260,78 @@ fabric.util.object.extend(fabric.IText.prototype, /** @lends fabric.IText.protot
 
   /**
    * Copies selected text
-   * @param {Event} e Event object
+   * @param {ClipboardEvent} e Event object
+   * @returns {boolean} handle event
    */
-  copy: function() {
+  setClipboardData: function (e) {
+    e.preventDefault();
     if (this.selectionStart === this.selectionEnd) {
-      //do not cut-copy if no selection
-      return;
+      e.clipboardData.effectAllowed = 'none';
+      return false;
     }
+    var clipboardData = e.clipboardData;
+    var value = this.getSelectedText();
+    clipboardData.effectAllowed = 'all';
+    clipboardData.setData('text/plain', value);
+    clipboardData.setData('application/fabric', JSON.stringify({
+      value: value,
+      styles: this.getSelectionStyles(this.selectionStart, this.selectionEnd, true)
+    }));
+    return true;
+  },
 
-    fabric.copiedText = this.getSelectedText();
-    if (!fabric.disableStyleCopyPaste) {
-      fabric.copiedTextStyle = this.getSelectionStyles(this.selectionStart, this.selectionEnd, true);
-    }
-    else {
-      fabric.copiedTextStyle = null;
-    }
-    this._copyDone = true;
+  /**
+    * Copies selected text
+    * @param {ClipboardEvent} e Event object
+    */
+  copy: function (e) {
+    this.setClipboardData(e) && this.fire('copy', { e: e });
   },
 
   /**
-   * Pastes text
-   * @param {Event} e Event object
+   * Cuts selected text
+   * @param {ClipboardEvent} e Event object
    */
-  paste: function() {
-    this.fromPaste = true;
+  cut: function (e) {
+    if (this.setClipboardData(e)) {
+      //  remove selection
+      this.insertChars('', null, this.selectionStart, this.selectionEnd);
+      this.selectionEnd = this.selectionStart;
+      this.hiddenTextarea && (this.hiddenTextarea.value = this.text);
+      this._updateTextarea();
+      this.fire('cut', { e: e });
+      this.fire('changed', { index: this.selectionStart, action: 'cut' });
+      this.canvas.fire('text:changed', { target: this });
+      this.canvas.requestRenderAll();
```

Comment:
this before was happening from the cut event naturally? why that is not more true?

Reply:
Was it? you mean the browser did it?

Reply:
I see. It did but I've decided to prevent default and handle it internally. Makes more sense IMO. And crucial for customization/overriding.